src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -20,26 +26,54 @@
 
 @RestController
 @RequestMapping("/api/books")
+@Api
 public class BookController {
 
     @Autowired
     private BookRepository bookRepository;
 
     @GetMapping("/{author}")
-    public Book findByAuthor(@PathVariable String author) {
+    @ApiOperation(value = "Given an author returns the book", response = Book.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, response = Book.class, message = "Returns the first book for that author"),
+            @ApiResponse(code = 404, message = "The book was not found")
+    })
+    public Book findByAuthor(
+            @ApiParam(value = "The author's name", required = true)
+            @PathVariable String author
+    ) {
         return bookRepository
                 .findFirstByAuthor(author)
                 .orElseThrow(BookNotFoundException::new);
     }
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public Book create(@RequestBody Book book) {
+    @ApiOperation(value = "Recieves a book to create", response = Book.class)
+    @ApiResponses(value={
+            @ApiResponse(code = 201, response = Book.class, message = "The book was created"),
+            @ApiResponse(code = 400, message = "There was a problem with the recieved data")
+    })
+    public Book create(
+            @ApiParam(value = "The book to be created.", required = true)
+            @RequestBody Book book
+    ) {
         return bookRepository.save(book);
     }
 
     @PutMapping("/{id}")
-    public Book update(@RequestBody Book book, @PathVariable Long id) {
+    @ApiOperation(value = "Receives an ID and a book to update", response = Book.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, response = Book.class, message = "The book was found and updated"),
+            @ApiResponse(code = 400, message = "The book id to update does not match the book id sent with data"),
+            @ApiResponse(code = 404, message = "Book not found")
+    })
+    public Book update(
+            @ApiParam(value = "The book to be updated", required = true)
+            @RequestBody Book book,
+            @ApiParam(value = "The book ID", required = true, example = "0")
```

Comment:
I found a bug with swagger version when there is no example setted for numeric params.
https://github.com/springfox/springfox/issues/2265
https://github.com/swagger-api/swagger-core/issues/2979

Reply:
Nice catch !